fvtest/tril/tril/ast.hpp

```diff
@@ -108,6 +137,34 @@ struct ASTValue {
      * b.get<double>(); // returns 4.5 as a double
      * b.get<int>();    // causes an assertion failure
      */
+
+#if __cplusplus <= 199711L
+    // Create create an equivalent implementation of std::enable_if.
+    template <bool, typename T = void>
+    struct enable_if {};
```

Comment:
It doesn't (at least I couldn't get it to work on GCC 4.4.6 [also, notice it says *at least as far back as GCC 4.4.6] by default).

But more importantly one of the original reasons why are doing this to begin with, to quote @fjeremic was 

>In particular attempting to build Tril on Linux on z with g++ 4.4.6 will not work due to missing C++11 support.

So instead of getting it to compile with some particular version of GCC, this patch is supposed to remove/work around the C++11 only features. And as far as I know, std::enable_if was "officially" only added since C++11.

Reply:
Did you end up compiling Tril (You should check the command lines) w/o any C++11 support? Or just whatever comes out of GCC 4.4.6 (i.e., `-std=c++0x`)?

Reply:
Can confirm, doesn't work.